src/Classes/CategoryHydrator.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Furniture\Classes;
+
+use \PDO;
+
+class CategoryHydrator
+{
+    public static function fetchAllCategories(PDO $db): array
+    {
+        $query = $db->prepare('SELECT `category_ID` AS `id`, `name`, COUNT(`name`) AS `products` FROM `products` `p` LEFT JOIN `categories` `c` ON `c`.`id` = `p`.`category_ID` GROUP BY `category_ID`');
```

Comment:
this query feels like its the wrong way round. you have the primary (first table referenced in the query) as `products` but your actually selecting category data. it doesnt change the functionality just feels wrong, makes it harder to figure out what the query is actually doing